rio/src/RIO/Prelude/Types.hs

```diff
@@ -0,0 +1,347 @@
+module RIO.Prelude.Types
```

Comment:
I think this needs to be added to `other-modules`, right? In fact, to ensure all modules are properly listed, perhaps we should stop using hpack's auto-detection for this package.

Reply:
Nah, that's the opposite of my intentions. Adding it to `other-module` will make the `RIO.Prelude` module bloated again, cause all types and classes and all their instances will move there. I want to avoid precisely that problem and for that reason kept the types in a separate module that can explored separately when one is looking for information about types, classes and instances.

Reply:
I do very much dislike `hpack`s auto detection feature. Quite often it messes with my workflow.

Reply:
OK, let's stop using autodetection then, that way there's no ambiguity when merging this PR which modules are supposed to be exposed and which ones hidden.

Reply:
Done